components/sc_cover/cover.py

```diff
@@ -38,6 +40,8 @@ async def to_code(config):
     await cg.register_component(var, config)
     await cover.register_cover(var, config)
 
+    cg.add(var.set_setup_delay(config[CONF_SETUP_DELAY]))
```

Comment:
This will throw and error if setup_delay is not set.

```suggestion
    setup_delay = config.get(CONF_SETUP_DELAY)
    if setup_delay:
        cg.add(var.set_setup_delay(setup_delay))
```

Reply:
Oops, yes, I understand, I'll apply this.